src/table/matrix_table.cpp

```diff
@@ -244,6 +244,8 @@ MatrixServerTable<T>::MatrixServerTable(integer_t num_row, integer_t num_col) :
 
 template <typename T>
 void MatrixServerTable<T>::ProcessAdd(const std::vector<Blob>& data) {
+  if (data.size() == 0) return;
```

Comment:
This should be controlled in Server::ProcessAdd. Make sure server will not reply when data is empty.